lib/gauge_polyakov_loop.cu

```diff
@@ -0,0 +1,276 @@
+#include <gauge_field.h>
+#include <gauge_tools.h>
+#include <instantiate.h>
+#include <tunable_nd.h>
+#include <tunable_reduction.h>
+#include <kernels/gauge_polyakov_loop.cuh>
+
+namespace quda {
+
+  template<typename Float, int nColor, QudaReconstructType recon>
+  class GaugeInsertTimeslice : public TunableKernel2D {
+    GaugeField &u;
+    const GaugeField &s;
+    int timeslice;
+    unsigned int minThreads() const { return s.LocalVolumeCB(); }
+
+  public:
+    GaugeInsertTimeslice(GaugeField &u, const GaugeField &s, int timeslice) :
+      TunableKernel2D(u, 2),
+      u(u),
+      s(s),
+      timeslice(timeslice)
+    {
+      strcat(aux, ",4d_vol=");
+      strcat(aux, u.VolString());
+
+      apply(device::get_default_stream());
+    }
+
+    void apply(const qudaStream_t &stream)
+    {
+      TuneParam tp = tuneLaunch(*this, getTuning(), getVerbosity());
+      GaugeInsertTimesliceArg<Float, nColor, recon> arg(u, s, timeslice);
+      launch<InsertTimeslice>(tp, stream, arg);
+    }
+
+    long long flops() const
```

Comment:
Can delete this function, since the virtual parent already returns 0

Reply:
Done locally, will push soon

Reply:
https://github.com/lattice/quda/commit/4a893d83c4c5b4c15c9e0d07f8cc3cd1a0e1955a